### DISCUSSION TITLE
try catch isnt working as expected, with fluentmigrator

### DISCUSSION THREAD

**Discussion by RamkiUIPath** [fluentmigrator/fluentmigrator#1537] [Q&A] [answered]
We are using fluentmigrator and sql server 2017. Below piece of code throws exception but catch block is not catching.

This throws exception because the command ADD SENSITIVITY CLASSIFICATION is not available in sql server 2017, it was introduced in 2019.

Execute is of type : IExecuteExpressionRoot

```
        public void AddSensitivityClassificationSql(string schema, string table, string column, string label, string informationType, string rank)
        {
            try
            {
                Execute.Sql($@"ADD SENSITIVITY CLASSIFICATION TO [{schema}].[{table}].[{column}] WITH (LABEL = '{label}', INFORMATION_TYPE = '{informationType}', RANK = {rank})");
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.ToString());
            }
        }
```

Because of this, program crashes. Can someone help me understand how i can catch this exception?

**Comment by fubar-coder** [comment, marked as answer]
The SQL statement isn't executed here. It's just tasked for execution later.

The workaround is to use the `Execute.WithConnection` which provides the DB connection and transaction and move the try/catch into the provided action.
